sklearn/manifold/t_sne.py

```diff
@@ -736,11 +738,9 @@ def _fit(self, X, skip_num_points=0):
                              "the metric or precomputed distances given "
                              "as X are not correct")
 
-        # Degrees of freedom of the Student's t-distribution. The suggestion
-        # degrees_of_freedom = n_components - 1 comes from
-        # "Learning a Parametric Embedding by Preserving Local Structure"
-        # Laurens van der Maaten, 2009.
-        degrees_of_freedom = max(self.n_components - 1.0, 1)
```

Comment:
We need to retain this behaviour by default. I.e. make `degrees_of_freedom=None` by default, and handle that case with this logic.